diff --git videostreaming/play.py videostreaming/play.py
index 8156e8a..f6fc15f 100755
--- videostreaming/play.py
+++ videostreaming/play.py
@@ -11,10 +11,11 @@ import gst
 
 import sys, argparse
 
-import player, player_gui
-import utils
-from audio_src import AudioSrc
-from video_src import VideoSrc
+from ndnvideo import player, player_gui
+from ndnvideo import utils
+
+from ndnvideo.audio_src import AudioSrc
+from ndnvideo.video_src import VideoSrc
 
 class GstPlayer(player.GstPlayer):
 	__pipeline = """
