diff --git csrc/lib/dir.mk csrc/lib/dir.mk
index 6489299..145eec7 100644
--- csrc/lib/dir.mk
+++ csrc/lib/dir.mk
@@ -78,7 +78,7 @@ shlib: $(SHLIBNAME)
 lib: libccn.a
 
 test: default keystore_check encodedecodetest ccnbtreetest
-	./encodedecodetest -o /dev/null
+	./encodedecodetest -k $(CCNX_DIR)/.ccnx_keystore -o /dev/null
 	./ccnbtreetest
 	./ccnbtreetest - < q.dat
 	rm -R _bt_*
@@ -87,12 +87,12 @@ dtag_check: _always
 	@./gen_dtag_table 2>/dev/null | diff - ccn_dtag_table.c | grep '^[<]' >/dev/null && echo '*** Warning: ccn_dtag_table.c may be out of sync with tagnames.cvsdict' || :
 
 keystore_check: ccn_initkeystore.sh
-	test -f "$$HOME/.ccnx/.ccnx_keystore" || $(MAKE) -f dir.mk new_keystore
+	test -f "$(CCNX_DIR)/.ccnx_keystore" || $(MAKE) -f dir.mk new_keystore
 
 new_keystore:
 	@echo === CCNx Keystore not found in your home directory
 	@echo === I will create one for you now '(^C to abort)'
-	sleep 1 && sh ccn_initkeystore.sh && sleep 3 && mv .ccnx "$$HOME"
+	sleep 1 && sh ccn_initkeystore.sh && sleep 3 && mkdir -p "$(CCNX_DIR)/" && mv .ccnx/.ccnx_keystore "$(CCNX_DIR)/"
 
 libccn.a: $(LIB_OBJS)
 	ar crus $@ $(LIB_OBJS)
@@ -113,7 +113,7 @@ skel_decode_test: skel_decode_test.o
 basicparsetest: basicparsetest.o libccn.a
 	$(CC) $(CFLAGS) -o $@ basicparsetest.o $(LDLIBS) $(OPENSSL_LIBS) -lcrypto
 
-encodedecodetest: encodedecodetest.o
+encodedecodetest: encodedecodetest.o libccn.a
 	$(CC) $(CFLAGS) -o $@ encodedecodetest.o $(LDLIBS) $(OPENSSL_LIBS) -lcrypto
 
 ccn_digest.o:
@@ -149,16 +149,16 @@ ccn_fetch.o:
 ccn_verifysig.o:
 	$(CC) $(CFLAGS) $(OPENSSL_CFLAGS) -c ccn_verifysig.c
 
-ccn_verifysig: ccn_verifysig.o
+ccn_verifysig: ccn_verifysig.o libccn.a
 	$(CC) $(CFLAGS) -o $@ ccn_verifysig.o $(LDLIBS) $(OPENSSL_LIBS) -lcrypto
 
 signbenchtest.o:
 	$(CC) $(CFLAGS) $(OPENSSL_CFLAGS) -c signbenchtest.c
 
-signbenchtest: signbenchtest.o
+signbenchtest: signbenchtest.o libccn.a
 	$(CC) $(CFLAGS) -o $@ signbenchtest.o $(LDLIBS) $(OPENSSL_LIBS) -lcrypto 
 
-ccndumppcap: ccndumppcap.o
+ccndumppcap: ccndumppcap.o libccn.a
 	$(CC) $(CFLAGS) -o $@ ccndumppcap.o $(LDLIBS) $(OPENSSL_LIBS) -lcrypto -lpcap
 
 ccnbtreetest.o:
