diff --git csrc/configure csrc/configure
index 962d76a..43ec3ea 100755
--- csrc/configure
+++ csrc/configure
@@ -39,7 +39,7 @@ TryAppend "`uname -sr | sed -e 's/ /-/g'`"
 TryAppend "local"
 TryScript "`uname`" "$@"
 TryScript "local" "$@"
-ProvideDefault CPREFLAGS = -I../include
+ProvideDefault CPREFLAGS = -I../include ${CFLAGS:-}
 ProvideDefault PCAP_PROGRAMS = ccndumppcap
 ProvideDefault RESOLV_LIBS = -lresolv
 ProvideDefault INSTALL_BASE = ${INSTALL_BASE:-/usr/local}
